Cython/Compiler/Nodes.py

```diff
@@ -759,8 +759,10 @@ def analyse(self, return_type, env, nonempty=0, directive_locals=None, visibilit
                 else:
                     self.exception_value = self.exception_value.analyse_types(env).coerce_to(
                         return_type, env).analyse_const_expression(env)
-                    exc_val = self.exception_value.get_constant_c_result_code()
-                    if exc_val is None:
+                    self.exception_value.calculate_constant_result()
+                    exc_val = self.exception_value.constant_result
+                    from .ExprNodes import constant_value_not_set
+                    if exc_val is None or exc_val is constant_value_not_set:
```

Comment:
Isn't this something that `ConstNode` could do? That would avoid the local import. Something like `ExprNode.as_exception_value()`, where the default implementation would call `error()` and subclasses could override it.

We should also make sure that this works for enum values etc., anything that is usable as  exception value.